concepts/eventing-code-guidelines.md

```diff
@@ -0,0 +1,1147 @@
+1. [ginkgo](https://github.com/onsi/ginkgo)
+-->
+
+**Assertion libraries**:
+
+<!-- voting options for assertion libraries in unit tests:
+Right now we use both: testing.T and gomega. We should reach a consensus here to have consistency.
+
+1. testing.T
+2. [stretchr/testify](https://github.com/stretchr/testify)
+3. [onsi/gomega](https://github.com/onsi/gomega)
+-->
+
+Use [`gomega`](https://github.com/onsi/gomega) for writing test **assertions** in controller integration tests. Do not use [stretchr/testify](https://github.com/stretchr/testify) unless you have a reason for it.
+
+**Mocking libraries**:
+
+Use [stretchr/testify/mock](https://github.com/stretchr/testify#mock-package) in combination with [vektra/mockery](https://github.com/vektra/mockery) for generating mocks or create your own mock by implementing the corresponding interface.
+
+<!-- voting options:
+1. [stretchr/testify](https://github.com/stretchr/testify#mock-package) with [vektra/mockery](https://github.com/vektra/mockery) // proposed by guidelines and already used by eventing-controller
+2. gomock aka [golang/mock](https://github.com/golang/mock)
+3. [ernesto-jimenez/goautomock](https://github.com/ernesto-jimenez/goautomock)
+4. [MarvinJWendt/testza](https://github.com/MarvinJWendt/testza)
+5. [gojuno/minimock](https://github.com/gojuno/minimock)
+-->
+
+### Structured Logging
+
+Both [logrus](https://github.com/sirupsen/logrus) and [zap](https://github.com/uber-go/zap) are widely used. logrus has the **advantage** and **disadvantage** at the same time of being a **drop-in** replacement for the **stdlib**.
+logrus supports structured logging but does not enforce using it. 
+Citing from the [logrus github page](https://github.com/sirupsen/logrus), it is clear that zap is the modern alternative to logrus:
+> Logrus is in maintenance-mode. We will not be introducing new features. It’s simply too hard to do in a way that won’t break many people’s projects, which is the last thing you want from your Logging library (again...).
+
+> Many fantastic alternatives have sprung up. Logrus would look like those, had it been re-designed with what we know about structured logging in Go today. Check out, for example, Zerolog, Zap, and Apex.
+
+For these reasons, `uber-go/zap` is the **preferred** structured logging library.
+
+Furthermore, consider using `github.com/kyma-project/kyma/common/logging/logger`, which provides further **abstraction** over `uber-go/zap`.
+
+<!-- voting options:
+1. [ueber-go/zap](https://github.com/uber-go/zap) // proposed by guidelines
+2. [logrus](https://github.com/sirupsen/logrus) 
+-->
+
+#### See also
+
+- [Kyma Logging Proposal](https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md)
+
+## Documentation guidelines
+
+Code is read many times but sometimes only written once. Therefore you should always make sure that you follow these guidelines.
+
+**Guidelines**:
```

Comment:
is the order of steps important, or can those guidelines apply in any order (I suspect the order doesn't matter). If it doesn't matter, I'd suggest an unordered list with bullets/dashes.

Reply:
Order is not important. Will change it as proposed :)